source/agora/common/crypto/Key.d

```diff
@@ -81,14 +88,19 @@ public struct KeyPair
     /// Generate a new, random, keypair
     public static KeyPair random () @nogc
     {
-        SecretKey sk;
-        PublicKey pk;
+        BitBlob!(crypto_sign_ed25519_SECRETKEYBYTES * 8) sk_data;
+        BitBlob!(crypto_core_ed25519_BYTES * 8) pk;
         Seed seed;
-        if (crypto_sign_keypair(pk[].ptr, sk[].ptr) != 0)
+        if (crypto_sign_keypair(pk[].ptr, sk_data[].ptr) != 0)
             assert(0);
-        if (crypto_sign_ed25519_sk_to_seed(seed[].ptr, sk[].ptr) != 0)
+        if (crypto_sign_ed25519_sk_to_seed(seed[].ptr, sk_data[].ptr) != 0)
             assert(0);
-        return KeyPair(pk, sk, seed);
+
+        Scalar x25519_sk;
+        if (crypto_sign_ed25519_sk_to_curve25519(cast(ubyte*)(x25519_sk[].ptr), sk_data[].ptr) != 0)
+            assert(0);
+        SecretKey sk = SecretKey(x25519_sk[]);
+        return KeyPair(PublicKey(pk[]), sk, seed);
```

Comment:
https://github.com/jedisct1/libsodium/blob/ae4add868124a32d4e54da10f9cd99240aecc0aa/src/libsodium/crypto_sign/ed25519/ref10/keypair.c#L69-L83

For reference